contracts/examples/bonding-curve/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+		only_owner!(self, "only owner may call this function");
+
+		let token = self.issued_token().get().identifier;
+		if self.issued_token().is_empty() {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				token.as_esdt_identifier(),
+				b" was not issued yet",
+			])));
+		}
+
+		require!(
+			self.supply_type().get() != SupplyType::Limited
+				|| self.max_supply().get() > self.supply().get(),
+			"Maximum supply limit reached!"
+		);
+
+		require!(
+			self.supply_type().get() != SupplyType::Limited
+				|| self.max_supply().get() >= self.supply().get() + amount.clone(),
+			"Minting will exceed the maximum supply limit!"
+		);
+
+		let caller = self.blockchain().get_caller();
+		self.mint_started_event(&caller, &amount);
+
+		Ok(ESDTSystemSmartContractProxy::new_proxy_obj(self.send())
+			.mint(&token, &amount)
+			.async_call()
+			.with_callback(self.callbacks().mint_callback(&caller, &amount)))
+	}
+
+	#[callback]
+	fn mint_callback(
+		&self,
+		caller: &Address,
+		amount: &Self::BigUint,
+		#[call_result] result: AsyncCallResult<()>,
+	) {
+		match result {
+			AsyncCallResult::Ok(()) => {
+				self.mint_success_event(caller);
+				self.supply().update(|supply| *supply += amount);
+				self.balance().update(|balance| *balance += amount);
+			},
+			AsyncCallResult::Err(message) => {
+				self.mint_failure_event(caller, message.err_msg.as_slice());
+			},
+		}
+	}
+
+	// endpoints
+
+	fn check_sell_requirements(
+		&self,
+		token: &TokenIdentifier,
+		amount: &Self::BigUint,
+	) -> SCResult<()> {
+		let issued_token = self.issued_token().get().identifier;
+		if &issued_token != token {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				b"Only ",
+				token.as_esdt_identifier(),
+				b" tokens accepted",
+			])));
+		}
+		if self.balance().get() < amount.clone() {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				b"Contract does not have enough ",
+				token.as_esdt_identifier(),
+				b". Please try again once more is minted.",
+			])));
+		}
+		require!(
+			amount > &Self::BigUint::zero(),
+			"Must pay more than 0 tokens!"
+		);
+		Ok(())
+	}
+
+	fn check_buy_requirements(
+		&self,
+		token: &TokenIdentifier,
+		amount: &Self::BigUint,
+	) -> SCResult<()> {
+		let exchanging_token = &self.exchanging_token().get().identifier;
+		if exchanging_token != token {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				b"Only ",
+				exchanging_token.as_esdt_identifier(),
+				b" tokens accepted",
+			])));
+		}
+		require!(
+			amount > &Self::BigUint::zero(),
+			"Must pay more than 0 tokens!"
+		);
+		Ok(())
+	}
+	fn get_curve_arguments(self) -> CurveArguments<Self::BigUint> {
+		CurveArguments {
+			supply_type: self.supply_type().get(),
+			max_supply: self.max_supply().get(),
+			current_supply: self.supply().get(),
+			balance: self.balance().get(),
+		}
+	}
+
+	fn get_function(&self) -> LinearFunction<Self::BigUint> {
+		LinearFunction {
+			a: Self::BigUint::from(3u64),
```

Comment:
Extract "3" and "5" as constants, something like DEFAULT_CONSTANT_COEFFICIENT and DEFAULT_LINEAR_COEFFICIENT.